### PR TITLE
[dynamo] Continue on fbgemm import fail

### PR DESCRIPTION
**Pull request by anijain2305**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #107496
* #107645
* #107634
* __->__ #107622

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov

### CODE REVIEW

File: torch/_dynamo/variables/user_defined.py [hunk 1]
```diff
@@ -570,7 +570,7 @@ class KeyedJaggedTensorVariable(UserDefinedObjectVariable):
     def is_matching_object(obj):
         try:
             from torchrec.sparse.jagged_tensor import KeyedJaggedTensor
-        except ImportError:
+        except (ImportError, AttributeError):
```
Comment (voznesenskym) [inline torch/_dynamo/variables/user_defined.py:573]: How

**Review by voznesenskym — APPROVED** [review]

File: torch/_dynamo/variables/user_defined.py (hunk 1 above)
Comment (anijain2305) [reply torch/_dynamo/variables/user_defined.py:573]: This happens when you have torchrec but some old version

Comment (ezyang) [reply torch/_dynamo/variables/user_defined.py:573]: The easiest way to reproduce this is accidentally install fbgemm-gpu-nightly at the same time as a source FBGEMM install, or perhaps with an incompatible source build of pytorch

**Comment by ezyang** [comment]
image

**Comment by voznesenskym** [comment]
> image

This is not that hard to do, actually

**Pull request closed without merging**